client/my-sites/checkout/checkout-thank-you/pending.tsx

```diff
@@ -55,21 +66,20 @@ function CheckoutPending( { orderId, siteSlug, redirectTo }: CheckoutPendingProp
 		if ( transaction ) {
 			const { processingStatus } = transaction;
 
-			if ( ORDER_TRANSACTION_STATUS.SUCCESS === processingStatus ) {
+			if ( SUCCESS === processingStatus ) {
 				const { receiptId } = transaction;
 
 				didRedirect.current = true;
+				const redirectUrl = redirectTo || `/checkout/thank-you/${ siteSlug }/${ receiptId }`;
```

Comment:
Previously, the `redirectTo` prop could contain a URL with the word "pending" and it would be replaced by the receipt ID. I don't believe this was intentional or ever used and it's risky because we don't know where the word "pending" might appear in a URL (eg: in the hostname) and this could therefore create invalid links.

Reply:
It turns out that the word `pending` as a placeholder was intentional! Or at least, it had a role. The backend code that injects the order ID into the URL also strips the `:receiptId` placeholder if no receipt yet exists. In that case, the URL will end with `/pending`, which the code removed here was treating as a placeholder itself. This should be fixed by https://github.com/Automattic/wp-calypso/pull/65213